modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/table/Table.tsx

```diff
@@ -750,7 +765,28 @@ function Table({
 
 		const storedFDSFieldsOrder = responseJSON?.fdsFieldsOrder;
 
+		const responseFields = await fetch(
+			`${API_URL.FDS_FIELDS}?filter=(${OBJECT_RELATIONSHIP.FDS_VIEW_FDS_FIELD_ID} eq '${fdsView.id}')&nestedFields=${OBJECT_RELATIONSHIP.FDS_VIEW_FDS_FIELD}`
+		);
+
+		if (!responseFields.ok) {
+			openDefaultFailureToast();
+
+			return null;
+		}
+
+		const responseFieldsJSON = await responseFields.json();
+
+		const storedFDSFields = responseFieldsJSON?.items;
+
```

Comment:
I don't think we need to fetch storedFields from the backend at this point. This `updateFDSFieldsOrder()` is triggered when fields are re-ordered inside the table, so there should not be other task to do here other than persist the new order (lines  742-756) and update the internal state of the component by calling `setFDSFields(reorderFDSFields({fdsFieldsOrder,fdsFields})`

Reply:
I tried using fdsFields and not fetching from the backend but I got the following error.
I also tried to add the | null but that did not work. I'm not too sure what to make of it.

`Table.tsx(771,5): error TS2345: Argument of type '{ fdsFieldsOrder: string; fdsFields: IFDSField[] | null; }' is not assignable to parameter of type '{ fdsFieldsOrder: string; storedFDSFields: IFDSField[]; }'.
  Object literal may only specify known properties, and 'fdsFields' does not exist in type '{ fdsFieldsOrder: string; storedFDSFields: IFDSField[]; }'.
`

Reply:
@fortunatomaldonado To prevent this ts error you just need to check that the value passed to the `storedFDSFields` exists. [L782](https://github.com/liferay-frontend/liferay-portal/pull/4053/files#diff-04631558821004cdb02882757dbe1f9e4bfdec979f6a4aa8ddec19bb94084e97R782)

```
if (fdsFields && storedFDSFieldsOrder && storedFDSFieldsOrder === fdsFieldsOrder) {
			const orderedFDSFields = reorderFDSFields({
				fdsFieldsOrder,
				storedFDSFields: fdsFields,
			});

			setFDSFields(orderedFDSFields);
```

This way, you don't need to fetch the fields.

Reply:
Thank you @juanjofgliferay!
I was able to make those changes and a feel others to resolve this.

Reply:
@dsanz I would say that it is fine with this change. We could move forward with it.